bwapi/bwdata.py

```diff
@@ -937,18 +937,16 @@ def _fill_params(self, name, startDate, data):
 
         for n in name_list:
             if isinstance(n, str):
-                if n not in self.ids:
+                if self.get_resource_id(n, check=True) == False:
                     logger.error(
-                        "Could not find {} with name {}".format(self.resource_type, n),
-                        self.ids,
+                        "Could not find {} with name {}".format(self.resource_type, n)
                     )
                 else:
-                    id_list.append(self.ids[n])
+                    id_list.append(self.get_resource_id(n))
             elif isinstance(n, int):
-                if n not in self.ids.values():
+                if self.get_resource_id(n, check=True) == False:
```

Comment:
NABD but I think this is a bit more Pythonic
```python
if not if self.get_resource_id(n, check=True):
```
(also on line 940)

Reply:
Fixed